de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java

```diff
@@ -161,6 +167,68 @@ protected SWTBotTreeItem openEditor(SWTBotTreeItem item) {
 		return newItem;
 	}
 	
+	/**
+	 * @param item Tree item that is beeing dragged
+	 * @param diagramEditor Graphiti diagram editor which the tree item is beeing dragged onto
+	 * 
+	 * We can only drop tree items to canvas elements. SWTBotGefViewer's canvas element is not 
+	 * capable of beeing accessed directly from external classes though. Further more SWTBotGefViewer's 
+	 * canvas member gets populated during runtime only. Hence we use reflections to get a hold on it here.
+	 */
+	protected void dragTreeItemToDiagramEditor(SWTBotTreeItem item, SWTBotGefEditor diagramEditor) {
+		SWTBotGefViewer viewer = diagramEditor.getSWTBotGefViewer();
+		SWTBotGefFigureCanvas canvas = null;		
+
+		for (Field f : viewer.getClass().getDeclaredFields()) {
+		    if ("canvas".equals(f.getName())) {
+		        f.setAccessible(true);
+		        try {
+		            canvas = (SWTBotGefFigureCanvas) f.get(viewer);
+		        } catch (IllegalArgumentException e) {

```

Comment:
Can we at least use `IllegalArgumentException | IllegalAccessException e` to make this more compact?